src/tritonserver.cc

```diff
@@ -2725,4 +2725,132 @@ TRITONSERVER_ServerInferAsync(
   return nullptr;  // Success
 }
 
+//
+// TRITONSERVER_MetricFamily
+//
+TRITONSERVER_Error*
+TRITONSERVER_MetricFamilyNew(
+    TRITONSERVER_MetricFamily** family, TRITONSERVER_MetricKind kind,
+    const char* name, const char* description, void* registry_ptr)
+{
+#ifdef TRITON_ENABLE_METRICS
+  const auto& registry =
+      reinterpret_cast<std::shared_ptr<prometheus::Registry>*>(registry_ptr);
+  // TODO: Remove this after review
+  // TODO: Verify passing registry_ptr from API caller is OK.
+  //       Calling GetRegistry() here did not add the new metrics to registry.
+  // const auto& registry = tc::Metrics::GetRegistry();
+  *family = reinterpret_cast<TRITONSERVER_MetricFamily*>(
+      new tc::MetricFamily(kind, name, description, *registry));
```

Comment:
Use try-catch in conjunction with my above comments to return errors when the MetricFamilu creation failed.